codegen/builtin_fs.go

```diff
@@ -708,6 +707,34 @@ func (dp DockerPush) Call(ctx context.Context, cln *client.Client, ret Register,
 		}
 	}
 
+	dockerAPI, err := DockerAPI(ctx)
+	if dockerAPI.Moby {
```

Comment:
It feels a little dangerous to dereference `dockerAPI` here even if the function returned an error.

Reply:
I guess we can also do:
```go
dockerAPI := DockerAPI(ctx)
if dockerAPI.Moby {
```

And then expose the error in `dockerAPI.Err`, what do you think?

Reply:
Yeah `dockerApi.Err` seems more reasonable here:
```
	dockerAPI := DockerAPI(ctx)
	if dockerAPI.Moby {
		// Return error only if dockerPush is using docker engine instead of buildkit.
		if dockerAPI.Err != nil {
			return Arg(ctx, 0).WithError(err)
		}
```